### ISSUE TITLE
Ingress Controller Deployment Fails As VMSS Fails To Update

### ISSUE THREAD

**Issue by barakAtSoluto** [Azure/aks-engine#803]
**This an ISSUE**
---

**What version of aks-engine?**:
Version: v0.32.0
GitCommit: c2b614849
GitTreeState: clean

---

**Kubernetes version**: v1.11.8

**What happened**: 
Nginx ingress controller with custom public IP fails as VMSS fails to update the set. 
VMSS error code: `ProvisioningState/failed/VMExtensionProvisioningError`

Message: `Multiple VM extensions failed to be provisioned on the VM. Please see the VM extension instance view for details.`

Logs from "kube-controller-manager" regarding the failure:
```
I0320 09:38:08.756687       1 azure_vmss.go:687] VirtualMachineScaleSetsClient.UpdateInstances for service (default/nginx-nginx-ingress-controller): scale set (k8s-devops-29607280-vmss) - updating, err=Long running operation terminated with status 'Failed': Code="VMExtensionProvisioningError" Message="Multiple VM extensions failed to be provisioned on the VM. Please see the VM extension instance view for details."
E0320 09:38:13.926011       1 azure_vmss.go:725] ensureHostsInVMSetPool() with scaleSet "k8s-devops-29607280-vmss" for service "default/nginx-nginx-ingress-controller" failed: Long running operation terminated with status 'Failed': Code="VMExtensionProvisioningError" Message="Multiple VM extensions failed to be provisioned on the VM. Please see the VM extension instance view for details."
E0320 09:38:13.926082       1 service_controller.go:219] error processing service default/nginx-nginx-ingress-controller (will retry): failed to ensure load balancer for service default/nginx-nginx-ingress-controller: Long running operation terminated with status 'Failed': Code="VMExtensionProvisioningError" Message="Multiple VM extensions failed to be provisioned on the VM. Please see the VM extension instance view for details."
I0320 09:38:13.927753       1 event.go:221] Event(v1.ObjectReference{Kind:"Service", Namespace:"default", Name:"nginx-nginx-ingress-controller", UID:"60065461-4a5e-11e9-a226-000d3a537433", APIVersion:"v1", ResourceVersion:"7862", FieldPath:""}): type: 'Warning' reason: 'CreatingLoadBalancerFailed' Error creating load balancer (will retry): failed to ensure load balancer for service default/nginx-nginx-ingress-controller: Long running operation terminated with status 'Failed': Code="VMExtensionProvisioningError" Message="Multiple VM extensions failed to be provisioned on the VM. Please see the VM extension instance view for details."
```
--

**What you expected to happen**:
The ingress controller deployment should have completed deploying a LB with a custom public IP attached and the following VMSS update have completed without failures.  

**How to reproduce it** (as minimally and precisely as possible):
1. Spine up a cluster with the following:
```
{
    "apiVersion": "vlabs",
    "properties": {
      "orchestratorProfile": {
        "orchestratorType": "Kubernetes",
        "orchestratorRelease": "1.11",
        "orchestratorVersion": "1.11.8",
        "kubernetesConfig": {
          "cloudProviderRateLimit": false,
          "kubeletConfig": {
            "--max-pods": "100"
          },
          "apiServerConfig": {
            "--admission-control":  "NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota,DenyEscalatingExec,AlwaysPullImages,ValidatingAdmissionWebhook,ResourceQuota"
          },
          "addons": [
            {
              "name": "cluster-autoscaler",
              "enabled": true,
              "config": {
                "min-nodes": "10",
                "max-nodes": "20"
              }
            },
            {
              "name": "smb-flexvolume",
              "enabled" : true,
              "containers": [
                {
                  "name": "keyvault-flexvolume",
                  "cpuRequests": "50m",
                  "memoryRequests": "10Mi",
                  "cpuLimits": "50m",
                  "memoryLimits": "10Mi"
                }
              ]
            },
            { 
              "name": "blobfuse-flexvolume",
              "enabled": true,
              "containers": [
                {
                  "name": "keyvault-flexvolume",
                  "cpuRequests": "50m",
                  "memoryRequests": "10Mi",
                  "cpuLimits": "50m",
                  "memoryLimits": "10Mi"
                }
              ]
            },
            {
              "name": "keyvault-flexvolume",
              "enabled" : true,
              "containers": [
                {
                  "name": "keyvault-flexvolume",
                  "cpuRequests": "50m",
                  "memoryRequests": "10Mi",
                  "cpuLimits": "50m",
                  "memoryLimits": "10Mi"
                }
              ]
            },
            {
              "name": "tiller",
              "enabled": false
            },
            {
              "name": "kubernetes-dashboard",
              "enabled": false
            }
          ]
        }
      },
      "masterProfile": {
        "count": 5,
        "vmSize": "Standard_D2_v2",
        "vnetSubnetID": "<cidr>",
        "vnetCidr": "<cidr>",
        "firstConsecutiveStaticIP": "<ip>",
        "storageProfile": "StorageAccount"
      },
      "agentPoolProfiles": [
        {
          "name": "devops",
          "count": 10,
          "vmSize": "Standard_D4_v2",
          "osType": "Linux",
          "availabilityProfile": "VirtualMachineScaleSets",
	        "diskSizesGB": [100],
          "vnetSubnetID": "<agent_subnet_id>",
          "acceleratedNetworkingEnabled": false,
          "storageProfile": "ManagedDisks"
        }
      ],
      "linuxProfile": {
        "adminUsername": "<user_name>",
        "ssh": {
          "publicKeys": [
          ]
        }
      },
      "servicePrincipalProfile": {
        "clientId": "",
        "secret": "",
        "objectId": ""
      },
      "aadProfile": {
        "clientAppID": "<guid>",
        "serverAppID": "<guid>",
        "tenantID": "<guid>",
        "adminGroupID": "<guid>"
      }
    }
  }
  
```

2. [Install helm](https://helm.sh/docs/using_helm/#install-helm) locally and on the cluster (after you have kubectl working with the cluster and helm installed locally run `helm init`).

3. Run -
`helm upgrade --install --debug nginx stable/nginx-ingress --version 0.29.1 -f values.yaml --wait`

Where  [Helm's values.yaml](https://github.com/helm/charts/blob/master/stable/nginx-ingress/values.yaml) is:

```
controller:
  livenessProbe:
    initialDelaySeconds: 30
  readinessProbe:
    initialDelaySeconds: 30 
  image:
    repository: quay.io/kubernetes-ingress-controller/nginx-ingress-controller
    tag: "0.22.0"
    pullPolicy: IfNotPresent
  replicaCount: 3
  extraArgs:
        default-ssl-certificate: <cert>
  stats:
    enabled: true
  resources:
    requests:
      cpu: "0.5"
      memory: "512Mi"
    limits:
      cpu: "2"
      memory: "2000M"
  config:
    large-client-header-buffers: "4 12k"
    proxy-buffer-size: "32k"
    vts-status-zone-size: "20m"
    disable-access-log: "false"
    ssl-ciphers: "<ciphers>"
    hide-headers: "<hideed-headers>"
    server-tokens: "False"
    ssl-protocols: "<ssl-protocols>"
    log-format-escape-json: "true"
    log-format-upstream: '<log_format>'
    http-snippet: |
      more_set_headers "Server: "; map $geoip_city_continent_code $is_eu_visit { default 0; EU 1; } 
    limit-conn-status-code: "429"
  metrics:
    enabled: true
    service:
      annotations:
        prometheus.io/port: "10254"
        prometheus.io/scrape: "true"
  autoscaling:
    enabled: true
    minReplicas: 3
    maxReplicas: 11
    targetCPUUtilizationPercentage: 100
    targetMemoryUtilizationPercentage: 100
  service:
    externalTrafficPolicy: "Local"
controller:
  service:
    loadBalancerIP: "<public_ip>"
    annotations:
      service.beta.kubernetes.io/azure-load-balancer-resource-group: <RG_name>
```

**Comment by jackfrancis** [comment]
Hi @barakAtSoluto, this was due to a new release of walinuxagent in the wild, and a bug in the way that our provisioning script handles that. It was addressed in #771 and #778 and those fixes were included in the v0.32.3 release:

https://github.com/Azure/aks-engine/releases/tag/v0.32.3

This bug only affected newly built clusters, so the fix is to re-build those cluster using v0.32.3

**Comment by palmerabollo** [comment]
Just in case it helps, I've seen this error in a traefik service with aks-engine 0.32.3:

>   Warning  CreatingLoadBalancerFailed  18m (x82 over 7h33m)     service-controller  Error creating load balancer (will retry): failed to ensure load balancer for service mynamespace/traefik: Long running operation terminated with status 'Failed': Code="VMExtensionProvisioningError" Message="Multiple VM extensions failed to be provisioned on the VM. Please see the VM extension instance view for details."
  Normal   EnsuringLoadBalancer        13m (x85 over 8h)        service-controller  Ensuring load balancer
